src/status_im/navigation/effects.cljs

```diff
@@ -117,11 +130,12 @@
 
 (defn open-modal
   [component]
-  (let [{:keys [options]} (get views/screens component)
-        sheet?            (:sheet? options)]
+  (let [{:keys [options name]} (get views/screens component)
+        sheet?                 (:sheet? options)]
     (if @state/dissmissing
       (reset! state/dissmissing component)
       (do
+        (set-view-id name)
```

Comment:
Please also add todo here, (Will be removed in https://github.com/status-im/status-mobile/issues/18811).
`component-did-appear-listener` should have called this function. We have to look into that and see why that didn't happen, and then we can remove this.